lua/orgmode/api/headline.lua

```diff
@@ -112,7 +113,8 @@ function OrgHeadline:priority_up()
   return self:_do_action(function()
     local headline = org.files:get_closest_headline()
     local current_priority = headline:get_priority()
-    local priority_state = PriorityState:new(current_priority)
+    local prio_range = config:get_priority_range()
+    local priority_state = PriorityState:new(current_priority, prio_range)
```

Comment:
I removed config from PriorityState, because I need to use PriorityState from Config to create the needed intermediate priorities in `Config:get_priorities()`. 
I first tried exactly, what you suggested, but this created a dependency cycle between Config and PriorityState, which doesn't work and throws a lot of errors instead.
Because PriorityState just needs this three simple values, it makes no sense to give it access to the full config, it is much cleaner to inject them - and it resolves the cycle.
Another way to avoid this little extra function `get_priority_range` would be to group the three config values for priorities directly in one single config value like this:
```lua
local DefaultConfig = {
  -- ...
  org_priorities = {
    org_priority_highest = "A",
    org_priority_lowest="C",
    org_priority_default="B",
  },
  -- ...
}
```

Reply:
Ah, makes sense, thanks!